### ISSUE TITLE
[feature request] Mapping GetItemID / get_item_id

### ISSUE THREAD

**Issue by jamesmurphy-mc** [pyimgui/pyimgui#332]
I'm noticing a pattern of doing this in some of my code
```
imgui_id = imgui.get_id(text)
imgui.whatever(text)

if imgui_id == _stored_imgui_id:
    ...
```
and I think what I really want is to get the id of the last item. I see that C++ Imgui has GetItemID that we have not mapped yet. I may take a stab at implementing this but before I do I wanted to get your opinion first. Is it as simple as wrapping the existing GetItemID like many other pyimgui calls, or could there be other issues that explain why it hasn't been mapped already (it is marked as "?")?

For background, the reason why I want to do this is because I prefer "is_item_clicked" behavior that returns true on the release event instead of on the press event, which requires remembering the last pressed item id to check if it is the same as the released item id, so I'm just writing that function myself.

**Comment by KinoxKlark** [comment]
It should be as easy as wrapping the existing `GetItemID.`

I marked functions in `internal` as `?` such that they don't count in the number of mapped/unmapped functions. The idea was that I didn't plane to map all internal functions but to map them as they were needed. (In other words, I did not want a 50% drop of the coverage just by adding the `internal` submodule ^^)

You should be able to map it without problems and mark it as `✓.`

**Comment by learn-more** [comment]
> It should be as easy as wrapping the existing `GetItemID.`
> 
> I marked functions in `internal` as `?` such that they don't count in the number of mapped/unmapped functions. The idea was that I didn't plane to map all internal functions but to map them as they were needed. (In other words, I did not want a 50% drop of the coverage just by adding the `internal` submodule ^^)
> 
> You should be able to map it without problems and mark it as `✓.`

If not done, it might make sense to document this `?` somewhere

**Comment by KinoxKlark** [comment]
It is, in the header of `internal.pxd`: 
https://github.com/pyimgui/pyimgui/blob/723022d87de8040d1c4a66f53288dcfd7d2274d3/imgui/internal.pxd#L4-L9

But I do agree that this could be done better and should also be documented in the `HACKING.md` or similar.

**Comment by KinoxKlark** [comment]
Yes, that one of the issues with exposing internals, they are not really ment to be used by user code. 

Indeed, item id corresponds to internal id for interactive elements, and text are not part of it.

I am not against implementing `internal.get_item_id()` as it may be of some help in some specific use case. But indeed, I am not sure it would help in what you are trying to achieve.

In `imgui_internal.h` there is a `ImGuiWindow::IDStack` which may be of some use. But I don't know to what extent.

This concept of reusing an identifier for the last imgui call sounds interesting and I would be open to some custom implementation of it implemented in cython.

I am also willing to add a custom `imgui.is_item_clicked_and_then_released()` (or whatever make sense as a name) in cython which encapsulate the pattern. It seems a reasonable functionality to have.
